### PR TITLE
Add/form block align tools

### PR DESCRIPTION
**Pull request by TimBroddin**
The proposed changes add the Align tool (Wide & Full Width) and address the following issue: [#65248](https://github.com/Automattic/wp-calypso/issues/65248)

| Before | After |
| -------|------|
|  CleanShot 2022-10-21 at 11 30 33@2x | CleanShot 2022-10-21 at 11 32 33@2x | 

Screen recording:
https://user-images.githubusercontent.com/528287/197164914-a4992809-51e4-4317-bf4a-9906a80cd5fb.mp4

#### Changes proposed in this Pull Request:
- add "inner-container" div - to make sure all inner blocks can be aligned on their own. Similarly to how the core Group block works
- add the Align support with Wide & Full Width options
- deprecate the current block version as version v3 and adjust fixtures

#### Jetpack product discussion
This PR is a part of the Expanding Jetpack Block design tooling for Full Site Editing project: pdDOJh-6-p2.

#### Does this pull request change what data or activity we track or use?
No

##### Main test
1. Create a new post or a page with the Form block
2. The Align tool (with the Wide and Full Width options) should display in the block toolbar
3. Try to adjust the alignment of the inner blocks, and maybe add some more. The align setting of each block should be respected (taking into consideration the Align setting of the Form block).
4. Experiment with the block, its Align tool and observe whether the changes apply correctly in the back end
5. Save the changes and review if the settings are correctly rendering in the front end as well

###### Backwards compatibility test
1. Before applying the changes in this PR, create a new post or a page with the Form block
2. Adjust the block settings, add inner blocks and save the page
3. Apply the code changes from this PR / patch
4. Reload the block editor and make sure the block is still working as expected. The block should validate correctly
5. Make additional changes to the block and save them
6. Review if the block is rendering correctly in both back end and front end

### CODE REVIEW

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -2592,8 +2592,19 @@ public static function parse( $attributes, $content ) {
...
+
+		if ( $has_alignwide ) {
+			$wrapper_classes = 'alignwide';
+		} elseif ( $has_alignfull ) {
+			$wrapper_classes = 'alignfull';
+		}
+
 		$r  = '';
-		$r .= "<div id='contact-form-$id'>\n";
+		$r .= "<div id='contact-form-$id' class='$wrapper_classes'>\n";
```
Comment (ivan-ottinger) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]: Have you tried to add the `alignwide` and `alignfull` classes using the block `supports` directly?

For instance, when we add the `supports` array when the block is registered in `register_contact_form_blocks()`, we can then access it through `apply_block_supports()`:

```diff
diff --git a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
index 099a38e1ed..a636271f78 100644
--- a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
+++ b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
@@ -398,6 +398,9 @@ class Grunion_Contact_Form_Plugin {
 			'jetpack/contact-form',
 			array(
 				'render_callback' => array( __CLASS__, 'gutenblock_render_form' ),
+				'supports'        => array(
+					'align' => array( 'wide', 'full' ),
+				),
 			)
 		);
 
@@ -2597,6 +2600,9 @@ class Grunion_Contact_Form extends Crunion_Contact_Form_Shortcode {
 		$has_alignfull   = preg_match( '/<div class=\"wp-block-jetpack-contact-form.*alignfull.*?">/m', $content );
 		$wrapper_classes = '';
 
+		$wrapper_attributes = \WP_Block_Supports::get_instance()->apply_block_supports();
+		var_dump( $wrapper_attributes );
+
 		if ( $has_alignwide ) {
 			$wrapper_classes = 'alignwide';
 		} elseif ( $has_alignfull ) {

```

In the example above we will get the following var dump:

```
array(1) { ["class"]=> string(44) "alignfull wp-block-jetpack-contact-form" }
```

where `alignfull` is coming directly from the block `support` and the alignment user selected in the editor. We did something similar in this PR: https://github.com/Automattic/jetpack/pull/22622.

This way we don't need to rely on `preg_match()`.

With that said, I might be missing something that you had in mind when you worked on this PR.

**Review by ivan-ottinger — COMMENTED** [review]
I tested the changes on Atomic and Jetpack site and the alignment tool is working as expected there - including cases where editing a Form block that had been added to the page before I applied the changes in this PR. 👍🏼👍🏼

The testing didn't work for me on WPCOM as when I applied the patch, the site crashed. But, this is likely because the PR needs a rebase:
Markup on 2022-11-04 at 14:42:29

---

When it comes to fixtures and block deprecation, the way I understand them (while also looking at the [Block Test Fixtures → Add a deprecation fixture doc](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/shared/test/block-fixtures.md#add-a-deprecation-fixture), is that when a new block version is introduced, we need to add a new set of deprecated fixture files.

For instance, when looking at our previous Form block PR: https://github.com/Automattic/jetpack/pull/22465, we added a new separate set of v2 fixture files that represented the previous version of the block (before changes).

In other words, I believe that this new PR should add a new set of v3 fixture files that will represent the block version before the alignment tool is added.

I found [this doc](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/) helpful when working with block deprecations.

---

I have also added a coupe of thoughts inline below.

---

Not sure if we will have a chance before my AFKs, but it might be nice pairing for this one and crush it together. 👨🏼‍💻👨🏼‍💻

**Comment by TimBroddin** [comment]
Duplicate of https://github.com/Automattic/jetpack/pull/27151
